src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Utilities/SymbolEquivalenceComparer.EquivalenceVisitor.cs

```diff
@@ -344,8 +344,7 @@ private bool HandleNamedTypesWorker(INamedTypeSymbol x, INamedTypeSymbol y, Dict
             {
                 Debug.Assert(GetTypeKind(x) == GetTypeKind(y));
 
-                if (x.IsDefinition != y.IsDefinition ||
-                    IsConstructedFromSelf(x) != IsConstructedFromSelf(y) ||
+                if (IsConstructedFromSelf(x) != IsConstructedFromSelf(y) ||
```

Comment:
this is the actual change.  the original check included something unnecessary (the IsDefinition check).  This was unnecessary, but not problematic in teh past.  however, with nullable this became an issue as `List<T!>` woud be not equal to `List<T>` due to the former having `IsDefinition=false`.  With symbol equivalence we don't want to consider these different.

Reply:
Is `IsDefinition` supposed to be different in those cases? It's unclear from the documentation on the property

```csharp
        /// <summary>
        /// Gets a value indicating whether the symbol is the original definition. Returns false
        /// if the symbol is derived from another symbol, by type substitution for instance.
        /// </summary>
```

Reply:
So I believe we decided that OriginalDefinition would always take you back to an unannotated type; for example if you have string? and string! then you want some way to get back to just "string".

Reply:
@CyrusNajmabadi I can never remember the precise definition of "equal" this class generally is trying to use. It's insensitive to nullability because for "is this symbol that symbol" we want it to work cross project/cross language?